syft/source/source.go

```diff
@@ -89,16 +89,28 @@ func ParseInputWithName(userInput string, platform string, detectAvailableImageS
 
 	// collect user input for downstream consumption
 	return &Input{
-		UserInput:                       userInput,
-		Scheme:                          scheme,
-		ImageSource:                     source,
-		Location:                        location,
-		Platform:                        platform,
-		Name:                            name,
-		autoDetectAvailableImageSources: detectAvailableImageSources,
+		UserInput:   userInput,
+		Scheme:      scheme,
+		ImageSource: source,
+		Location:    location,
+		Platform:    platform,
+		Name:        name,
 	}, nil
 }
 
+func defaultImageSourceFromConfig(defaultImageSource string) image.Source {
```

Comment:
nit: this isn't really coming from a config at this layer. This feels more like `parseDefaultImageSource`.